ops/jenkins/bfd-mgmt-synthea-generation-build/build-push.sh

```diff
@@ -1,31 +1,33 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
-# BUILD_CONTEXT_ROOT_DIR will return the directory where this script is located, and CONTEXT_DIR will be the
-# directory of the Docker build context. This way, this script can be called from _any_ directory and
-# there will be no issues
-BUILD_CONTEXT_ROOT_DIR="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")"
-readonly BUILD_CONTEXT_ROOT_DIR
+# BUILD_ROOT_DIR will return the directory where this script is located, and CONTEXT_DIR will be the
+# directory of the Docker build context. This way, this script can be called from _any_ directory
+# and there will be no issues
+BUILD_ROOT_DIR="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")"
+readonly BUILD_ROOT_DIR
+
+REPO_ROOT="$(cd "$BUILD_ROOT_DIR" && git rev-parse --show-toplevel)"
+readonly REPO_ROOT
 
-REPO_ROOT="$(git rev-parse --show-toplevel)"
 CONTEXT_DIR="${REPO_ROOT}/ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation"
 readonly CONTEXT_DIR
 
-# We want this script to be included in the Docker image while still living inside the build context,
-# so we'll pass the script as an arg into our build command
-DOCKER_ENTRYPOINT_SCRIPT="$(cat "${BUILD_CONTEXT_ROOT_DIR}/docker_entrypoint.sh")"
+# We want this script to be included in the Docker image while still living inside the build dir, so
+# we'll pass the script as an arg into our build command
+DOCKER_ENTRYPOINT_SCRIPT="$(cat "${BUILD_ROOT_DIR}/docker_entrypoint.sh")"
 readonly DOCKER_ENTRYPOINT_SCRIPT
 
 SYNTHEA_PROPERTIES_RAW_URL="https://raw.githubusercontent.com/synthetichealth/synthea/master/src/main/resources/synthea.properties"
 readonly SYNTHEA_PROPERTIES_RAW_URL
 
-SYNTHEA_PROPERTIES_FILE_DIR="$BUILD_CONTEXT_ROOT_DIR/src/main/resources"
+SYNTHEA_PROPERTIES_FILE_DIR="$CONTEXT_DIR/src/main/resources"
 readonly SYNTHEA_PROPERTIES_FILE_DIR
 
-SYNTHEA_MAPPING_FILES_DIR="$BUILD_CONTEXT_ROOT_DIR/src/main/resources/export"
+SYNTHEA_MAPPING_FILES_DIR="$CONTEXT_DIR/src/main/resources/export"
```

Comment:
Was the path wrong before?

Reply:
Yes. The `$BUILD_CONTEXT_ROOT_DIR` (or, rather, it's now `$BUILD_ROOT_DIR`) refers to the parent directory of the `build-push.sh` script, which relative to the root of the repository is `ops/jenkins/bfd-mgmt-synthea-generation-build`. However, on line 102 we pass `$CONTEXT_DIR` as the Docker build context directory, which is, relative to the root of the repository, `ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation`.

This meant that the `build-push.sh` script was pulling the Synthea mapping files into the `ops/jenkins/bfd-mgmt-synthea-generation-build` directory, but Docker was building the image from files within the `ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation` directory. So, the image was getting built without the necessary mapping files or `synthea.properties`, and so it was failing to run.

Reply:
Yes, this was adjusted in a prior PR but some testing must have been incorrect, since via testing it wasn't working. Mitch fixed the paths and it worked after the adjustment.